### ISSUE TITLE
Library

### ISSUE THREAD

**Issue by DrHugoAndres** [patmo141/odc_public#47]
I'm in the final stages of the Breath Sleep Splint, i'm validating protusion gauge! I would like to make a contribution in your repository.
I need to know how to create/increase tooth libraries, implants, abutmans and  to focus on increasing ODC functionality. I did not understand how you managed to divide your teeth into mesial, lingual, vestibular, is there a code for this? Can you show me?

**Comment by patmo141** [comment]
Open up one of the tooth library .blend files in the "data" folder  
![image](https://user-images.githubusercontent.com/2917743/32986858-231b8d98-cca9-11e7-843f-28cfb0c80edf.png)

**Comment by patmo141** [comment]
1)  First thing to notice is that the local orientation of each tooth has the following properies
-The origin is roughly at the center
-The Z axis points toward the Occlusal
-The X axis poinrts toward the Mesial
-This means that the Y axis will point in different directions for different quadrants (Toward the Facial for quadrants 1 and 3, and toward the palatal for quadrants 2 and 4.

![image](https://user-images.githubusercontent.com/2917743/32986919-852309ac-ccaa-11e7-904a-8c5c3940663b.png)

**Comment by patmo141** [comment]
2.  The teeth must be named with the FDI system.  16 is the upper right molar.
![image](https://user-images.githubusercontent.com/2917743/32986938-4a168dc4-ccab-11e7-8d93-6547cff40075.png)

**Comment by patmo141** [comment]
If you follow those first 2, then you library will work for tooth placement, waxup, tooth to curve and some other functions, but it will not work for  Crown/Bridge or Flexi Tooth.

**Comment by patmo141** [comment]
Topology and Vertex Groups

For you teeth to work with the crown and bridge tools (like seat to margin), the teeth have to be very stricly designed.  For this you will need to make a copy of the ODC_toot_libaray and then modify those teeth.

1.  Vertex Groups:
![vertex_groups](https://user-images.githubusercontent.com/2917743/32986967-060742e4-ccac-11e7-9ff6-8941c189b3de.gif)

2. Multires Modifier on level 3.  So you will start with the basic tooth library, and then modify/sculpt/snap the higher multires data levels to match your desired tooth shape.
![multires_modifier](https://user-images.githubusercontent.com/2917743/32986973-282a5eec-ccac-11e7-8612-649b2e40adb5.gif)

**Comment by DrHugoAndres** [comment]
Can I make a mirror modifier to sculpt only half of them? or for scan 3d retopolize and mirror the teeth?

**Comment by DrHugoAndres** [comment]
Its more easy for build the library!

**Comment by patmo141** [comment]
Yes, only sculpt half, and we mirror to get the others.  Don't use mirror modifier .  Yes you can retopologize scanned teeth by using shrink-wrap to adapt the original odc teeth to the scan data.

**Comment by DrHugoAndres** [comment]
In the right (Toward the Facial for quadrants 1 and 3) I sculpt them and the left ( toward the palatal for quadrants 2 and 4) I scale lower posterior teeth.
[odc_tooth_library_sculpted.zip](https://github.com/patmo141/odc_public/files/1492927/odc_tooth_library_sculpted.zip)

**Comment by DrHugoAndres** [comment]
[odc_tooth_library_no_sculpted.zip](https://github.com/patmo141/odc_public/files/1540766/odc_tooth_library_no_sculpted.zip)
I don't have a free time now.
We should do a Denture library for manufacturing the tooth and gengival
[odc_tooth_library_no_sculpted.zip](https://github.com/patmo141/odc_public/files/1540771/odc_tooth_library_no_sculpted.zip)
